tests/test_data_generators.py

```diff
@@ -177,6 +128,24 @@ def test_DataGeneratorPytorch():
     assert (np.array(counts) > 0.75).sum() > 0.22 * total
 
 
+def test_compute_validation_generator():
+    num_of_unique_inchikeys = 15
```

Comment:
We could add an extra test with saving and loading from pickle. So after creating the val_generator save it by pickling. Create the validation pairs for the original and the pickled one and check that the results are exactly the same.